release/downloadIstioCtl.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+#
+# Separate downloader for istioctl
+#
+# This file will be fetched as: curl -L https://git.io/getLatestIstioCtl | sh -
+#
+detect_os() {
+  OS=$(uname)
+  case $OS in
+    'Linux')
+      OSEXT='linux'
+      FILENAME='istioctl'
+      ;;
+    'WindowsNT')
+      OSEXT='win.exe'
+      FILENAME='istioctl.exe'
+      ;;
+    'Darwin') 
+      OSEXT='osx'
+      FILENAME='istioctl'
+      ;;
+    *) ;;
+  esac
+}
+
+get_version() {
+  if [ "x${ISTIO_VERSION}" = "x" ] ; then
+  ISTIO_VERSION=$(curl -L -s https://api.github.com/repos/istio/istio/releases/latest | \
+                  grep tag_name | sed "s/ *\"tag_name\": *\"\\(.*\\)\",*/\\1/")
+  fi
+}
+
+download_istioctl() {
+  NAME="istioctl-${OSEXT}"
+  URL="https://storage.googleapis.com/istio-artifacts/${ISTIO_VERSION}/artifacts/istioctl/${NAME}"
+  echo "${URL}"
+  echo "Downloading ${NAME} ..."  
+  curl -L "${URL}" -o "${FILENAME}"
+  BINDIR="$(pwd)"
```

Comment:
is this what we want? I am probably going to add this to run this from a random directory that may get deleted.
FWIW kubectl you just directly download a binary with `curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl`. Seems a bit cleaner than piping to sh which is [generally a bad idea](https://www.seancassidy.me/dont-pipe-to-your-shell.html). linkerd does the same but outputs it to something like ~/.linkerd/linkerd.

Not sure whats best, I am not a UX expert :slightly_smiling_face:

Reply:
Missed htis -good catch @howardjohn .  I'd recommend installing into $HOME/istioctl/istioctl.

and then ask the user to add $HOME/istioctl their PATH (alternatively they can copy the $HOME/istioctl binary to their path which is  typically what I do).

the chmod should probably be a thing in this script as well.

Reply:
@howardjohn, @sdake:

So, the approach here is basically in response to #11527. This issue is a feature request to provide a separate download for the `istioctl` bins. 

We previously discussed the use of the GCS URLs, but @linsun mentioned [here](https://github.com/istio/istio/issues/11527#issuecomment-481940785) that we should "obfuscate" the git commit and underlying storage platform by offering our users a `/getLatestIstioCtl` URL, just as we do with the `/getLatestIstio` URL, and have our users download `istioctl` with a piping to sh technique.

This is where the creation of this script originates from. Once this PR is merged, the idea was to have the `/getLatestIstioCtl` URL pointing to it. 

If I understand the suggestion here, this script should ultimately download the `istioctl` binary into `$HOME/istioctl/istioctl`, then prompt the user to add `$HOME/istioctl` to their path, correct? Though, isn't this still piping the script to sh? Do we want to do that at all? If not, then this PR potentially isn't needed. Sounds like we need consensus on that part of it, before modifying the script, correct?

Reply:
I do think we want 1 single cmd that works for different platforms to download istioctl.  I don't like the way k8s has it where it has a tab or different cmd to copy based on the platform.

Can we download it to $HOME/.istio/bin and guide user to add it to the path manually as part of the output?

Reply:
@linsun yes, we can modify the script do that. Just to be clear, if we are creating this 1 script that works for different platforms (mimicking what linkerd is doing with their CLI download), we're still piping the downloaded script to sh for execution. I'm totally ok with that, just want to make sure everyone is on board with it....

 /cc @howardjohn @sdake